### PR TITLE
2.3 dev - IOS-529 (work in progress) 

### PR DESCRIPTION
**Pull request by SilviuOdobescu**

### CODE REVIEW

File: AlfrescoApp/Model/Managers/RealmSyncHelper.m [hunk 1]
```diff
@@ -120,6 +120,26 @@ - (NSString *)syncIdentifierForNode:(AlfrescoNode *)node
     return syncIdentifier;
 }
 
+- (NSMutableArray *)syncIdentifiersForNodes:(NSArray *)nodes
+{
+    NSMutableArray *syncIdentifiers = [NSMutableArray array];
+    
+    for (AlfrescoNode *node in nodes)
+    {
+        [syncIdentifiers addObject:[self syncIdentifierForNode:node]];
```
Comment (mikeh) [inline AlfrescoApp/Model/Managers/RealmSyncHelper.m]: Is it worth performing a nil check before adding to the array? In theory `nodeRefWithoutVersionID` could return nil - unlikely, but still possible.

File: AlfrescoApp/Model/Managers/RealmSyncManager.m [hunk 2]
```diff
@@ -208,11 +209,49 @@ - (void)deleteNodeFromSync:(AlfrescoNode *)node withCompletionBlock:(void (^)(BO
...
 }
 
+- (void)saveDeletedFileBeforeRemovingFromSync:(AlfrescoDocument *)document
+{
+    NSString *contentPath = [self contentPathForNode:document];
+    NSMutableArray *syncObstableDeleted = [_syncObstacles objectForKey:kDocumentsDeletedOnServerWithLocalChanges];
+    
+    // copying to temporary location in order to rename the file to original name (sync uses node identifier as document name)
+    NSString *temporaryPath = [NSTemporaryDirectory() stringByAppendingPathComponent:document.name];
+    [self.fileManager copyItemAtPath:contentPath toPath:temporaryPath error:nil];
```
Comment (mikeh) [inline AlfrescoApp/Model/Managers/RealmSyncManager.m]: This doesn't seem to handle the case when a file already exists in the temp folder. We have existing code for those cases in `BaseInboundURLHandler` and `TextFileViewController`

File: AlfrescoApp/Model/Managers/RealmSyncHelper.m (hunk 1 above)
Comment (SilviuOdobescu) [inline AlfrescoApp/Model/Managers/RealmSyncHelper.m]: Indeed - added the check in code and will be available in the next commit.

File: AlfrescoApp/Model/Managers/RealmSyncManager.m (hunk 2 above)
Comment (SilviuOdobescu) [inline AlfrescoApp/Model/Managers/RealmSyncManager.m]: Will reuse the logic for this case as well.

**Pull request merged**